packages/smooth_app/lib/pages/folksonomy/folksonomy_card.dart

```diff
@@ -0,0 +1,146 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:smooth_app/generic_lib/design_constants.dart';
+import 'package:smooth_app/helpers/product_cards_helper.dart';
+import 'package:smooth_app/helpers/ui_helpers.dart';
+import 'package:smooth_app/pages/folksonomy/folksonomy_list.dart';
+import 'package:smooth_app/themes/constant_icons.dart';
+
+class FolksonomyCard extends StatefulWidget {
+  const FolksonomyCard(this.product);
+  final Product product;
+
+  @override
+  _FolksonomyCardState createState() => _FolksonomyCardState();
+}
+
+class _FolksonomyCardState extends State<FolksonomyCard> {
+  late Future<Map<String, ProductTag>> _productTagsFuture;
```

Comment:
Generally speaking, storing a `Future` is useless.
+ A `FutureBuilder` doesn't allow relaunching it and we're also mixing UI / logic.

Instead, you can have a look to the `ScanTagline`, with a `ChangeNotifier/ValueNotifier` managing everything and just passing the result to the UI